src/main/java/org/broadinstitute/hellbender/tools/exome/ACSCopyRatioModeller.java

```diff
@@ -129,21 +135,25 @@ public ACSCopyRatioModeller(final SegmentedModel segmentedModel) {
                 final int segment = i;
                 final int targetStartInSegment = dataCollection.getStartTargetInSegment(segment);
                 final int numTargetsInSegment = dataCollection.getNumTargetsInSegment(segment);
-                final Function<Double, Double> logConditionalPDF = newMean -> {
-                    double ll = 0.;
-                    for (int target = 0; target < numTargetsInSegment; target++) {
-                        final double coverage = dataCollection.getCoveragesInSegment(segment).get(target);
-                        if (!state.getOutlierIndicator(targetStartInSegment + target)) {
-                            ll -= normalTerm(coverage, newMean, state.variance());
+                if (numTargetsInSegment == 0) {
+                    means.add(Double.NaN);
+                } else {
+                    final Function<Double, Double> logConditionalPDF = newMean -> {
+                        double ll = 0.;
```

Comment:
Perhaps I have drunk the Java 8 Kool-Aid, but I have started thinking of `double/int <variable> = 0` followed by a `for` loop involving `<variable>` as a possible code smell.  IMO the following is more readable:
```
final Function<Double, Double> logConditionalPDF = newMean -> IntStream.range(0, numTargetsInSegment)
			.filter(target -> !state.getOutlierIndicator(targetStartInSegment + target))
			.map(target -> dataCollection.getCoveragesInSegment(segment).get(target))
			.map(coverage -> -normalTerm(coverage, newMean, state.variance()))
			.sum();
```